specs/core/1_custody-game.md

```diff
@@ -0,0 +1,483 @@
+
+For each `challenge` in `block.body.bit_challenges`, run the following function:
+
+```python
+def process_bit_challenge(state: BeaconState,
+                          challenge: BitChallenge) -> None:
+    # Verify challenge signature
+    challenger = state.validator_registry[challenge.challenger_index]
+    assert bls_verify(
+        message_hash=signed_root(challenge),
+        pubkey=challenger.pubkey,
+        signature=challenge.signature,
+        domain=get_domain(state, get_current_epoch(state), DOMAIN_BIT_CHALLENGES)
+    )
+    # Verify the challenger is not slashed
+    assert challenger.slashed is False
+    # Verify attestation
+    assert verify_standalone_attestation(state, convert_to_standalone(state, challenge.attestation))
+    # Verify the attestation is eligible for challenging
+    responder = state.validator_registry[challenge.responder_index]
+    min_challengeable_epoch = responder.exit_epoch - EPOCHS_PER_CUSTODY_PERIOD * (1 + responder.max_reveal_lateness)
+    assert min_challengeable_epoch <= slot_to_epoch(challenge.attestation.data.slot) 
+    # Verify the responder participated in the attestation
+    assert challenge.responder_index in attestation.validator_indices
+    # A validator can be the challenger or responder for at most one challenge at a time
+    for challenge_record in state.bit_challenge_records:
+        assert challenge_record.challenger_index != challenge.challenger_index
+        assert challenge_record.responder_index != challenge.responder_index
+    # Verify the responder subkey
+    assert verify_custody_reveal(CustodyReveal(
+        revealer_index=challenge.responder_index,
+        period=epoch_to_custody_period(slot_to_epoch(attestation.data.slot)),
+        subkey=challenge.responder_subkey,
+    ))
+    # Verify the chunk bits count
+    chunk_bits_count = get_crosslink_chunk_count(challenge.attestation)
+    verify_bitfield(challenge.chunk_bits, chunk_bits_count)
+    # Verify the xor of the chunk bits does not equal the custody bit
+    chunk_bits_xor = 0b0
+    for i in range(chunk_bits_count):
+        chunk_bits_xor ^ get_bitfield_bit(challenge.chunk_bits, i)
+    custody_bit = get_bitfield_bit(attestation.custody_bitfield, attestation.validator_indices.index(responder_index))
+    assert custody_bit != chunk_bits_xor
+    # Add new bit challenge record
+    state.bit_challenge_records.append(BitChallengeRecord(
+        challenge_index=state.challenge_index,
+        challenger_index=challenge.challenger_index,
+        responder_index=challenge.responder_index,
+        crosslink_data_root=challenge.attestation.crosslink_data_root,
+        chunk_bits=challenge.chunk_bits,
+        responder_subkey=challenge.responder_subkey,
+        deadline=get_current_epoch(state) + CUSTODY_RESPONSE_DEADLINE
+    ))
+    state.challenge_index += 1
+    # Postpone responder withdrawability
+    responder.withdrawable_epoch = FAR_FUTURE_EPOCH
```

Comment:
Does this not open a grieving vector whereby a responder's exit is delayed up to `CUSTODY_RESPONSE_DEADLINE` by repeatedly posing challenges?

Reply:
Vitalik is planning to revamp withdrawability. See item 23 in #675. To be discussed in today's call.